zerver/lib/emoji.py

```diff
@@ -36,12 +37,21 @@
     + f"))(?![^{terminal_symbols}])"
 )
 
+
+def get_colon_emoji(emoji: Match[str], emoticon: str) -> str:
+    emoticon_present = emoji.group(0)
+    if emoticon in emoji.string.split():
+        return EMOTICON_CONVERSIONS[emoticon_present]
+    return emoticon_present
+
+
 # Translates emoticons to their colon syntax, e.g. `:smiley:`.
 def translate_emoticons(text: str) -> str:
     translated = text
-
     for emoticon in EMOTICON_CONVERSIONS:
-        translated = re.sub(re.escape(emoticon), EMOTICON_CONVERSIONS[emoticon], translated)
+        translated = re.sub(
+            re.escape(emoticon), lambda emoji: get_colon_emoji(emoji, emoticon), translated
+        )
```

Comment:
Can you add a test for this logic change in this commit?  That'd make it a lot easier to see what bug is fixed and prevent it from regressing in the future.